src/renderer/components/Data/NewDatasetModal.tsx

```diff
@@ -21,44 +10,87 @@ import {
   ModalDialog,
   Sheet,
   Typography,
+  Box,
+  CircularProgress,
 } from '@mui/joy';
+import { PlusCircleIcon } from 'lucide-react';
 
 import * as chatAPI from '../../lib/transformerlab-api-sdk';
 
-const uppy = new Uppy().use(XHR, {
-  endpoint: '',
-});
-
-// uppy.on('complete', () => {
-//   console.log('Modal is open');
-// });
+const fetcher = (url) => fetch(url).then((res) => res.json());
 
 export default function DatasetDetailsModal({ open, setOpen }) {
   const [newDatasetName, setNewDatasetName] = useState('');
   const [showUploadDialog, setShowUploadDialog] = useState(false);
+  const [trainFileUploaded, setTrainFileUploaded] = useState(false);
+  const [evalFileUploaded, setEvalFileUploaded] = useState(false);
+  const [uploading, setUploading] = useState(false);
+  const [formData, setFormData] = useState(new FormData());
 
   // Reset newDatasetName when the modal is open/closed
   // useEffect(() => {
   //   setNewDatasetName('');
   // }, [open]);
+  const { data, isLoading } = useSWR(
+    chatAPI.Endpoints.Dataset.LocalList(),
+    fetcher
+  );
+  //Resetting state variables
+  const handleClose = () => {
+    setOpen(false);
+    setShowUploadDialog(false);
+    setNewDatasetName('');
+    setEvalFileUploaded(false);
+    setTrainFileUploaded(false);
+    setFormData(new FormData());
+  };
 
-  uppy?.getPlugin('XHRUpload')?.setOptions({
-    endpoint: chatAPI.Endpoints.Dataset.FileUpload(newDatasetName),
-    getResponseError: function getResponseError(responseText, response) {
-      try {
-        alert(JSON.parse(responseText)?.detail);
-      } catch (e) {
-        return responseText;
-      }
-    },
-  });
-
+  useEffect(() => {
+    //This needs to be in a useEffect because we need to wait for state variables to update
+    if (trainFileUploaded && evalFileUploaded) {
+      uploadFiles();
+    }
+  }, [trainFileUploaded, evalFileUploaded]);
+  const uploadFiles = async () => {
+    setUploading(true); //This is for the loading spinner
+    //Create the dataset before uploading
+    const response = await fetch(
+      chatAPI.Endpoints.Dataset.Create(newDatasetName)
+    );
+    const data = await response.json();
+    if (data.status == 'error') {
+      alert(data.message);
+    } else {
+      fetch(chatAPI.Endpoints.Dataset.FileUpload(newDatasetName), {
```

Comment:
This fetch call is asynchronous so the setUploading() call will happen immediately. Is this what we want, or should we do a callback .then() type thing?